### DISCUSSION TITLE
How to plot the Grain area map ?

### DISCUSSION THREAD

**Discussion by gskgs** [mtex-toolbox/mtex#1445] [Ask Anything] [answered]
Dear colleagues and Professors.

I am pretty new to MTEX. I would like to know[ how to plot the Grain area map ]shows difference in term of grain area . 

![222](https://user-images.githubusercontent.com/106585808/171134446-ebe07e32-2061-4265-b9db-06959c92170a.png)

**Comment by kilir** [comment, marked as answer]
Hi,
please have a look here: https://mtex-toolbox.github.io/Documentation.html
Cheers,
Rüdiger

**Reply by gskgs** [reply]
Thank you very much
I find a code
`% reconstruct grains
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',5*degree);

% remove small grains
ebsd(grains(grains.grainSize<=1)) = [];

% redo grain reconstruction
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',2.5*degree);

% smooth grain boundaries
grains = smooth(grains,5);
plot(grains,grains.area)`





I get the map 
![123](https://user-images.githubusercontent.com/106585808/171214365-c0a8ee0d-ba30-40b4-8cd4-77b03d1bedee.png)


And I want to merge grains to reconstruct parent grains, I need to get the area map of phase before twinning.
I use this code.

%calculate the grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'threshold',10*degree);
%Copy the GB information to a new variable
gB = grains.boundary
%To not include the border of the EBSD map in calculations, create a new variable for only the mg-mg segments
gB_NiNi = gB('Ni-superalloy','Ni-superalloy')
%which can then be plotted, colorised by misorientation if you want
figure;plot(gB_NiNi,gB_NiNi.misorientation.angle./degree,'linewidth',2)
mtexColorbar

cs=ebsd ('Ni-superalloy').CS
%define an extension twin (60 degrees reorientation) by axis and rotation
%for magnesium extension twin
h= Miller(1,1,1, cs)
twinning = orientation('axis',h,'angle',60*degree,cs,cs)
%Next we create a logical of the boundary segments within 5 degrees of the twin relationship we just defined.
isTwinning = angle(gB_NiNi.misorientation,twinning) < 5*degree;
%locate the segments that meet this condition
twinBoundary = gB_NiNi(isTwinning)
% plot the grains, and overlay the twinning boundaries
figure;plot(grains,grains.meanOrientation)
hold on
plot(twinBoundary,'linecolor','blue','linewidth',2,'displayName','twin boundary')
hold off
%to calculate percentages of twin boundary out of all boundary segments
100*length(twinBoundary)/length(gB_NiNi)
[mergedGrains,parentId] = merge(grains,twinBoundary);
% plot the merged grains
hold on
plot(mergedGrains.boundary,'linecolor','k','linewidth',2.5,'linestyle','-',...
  'displayName','merged grains')
hold off
% copy ebsd data into a new variable to not change the old data
ebsd_merged = ebsd;
% update the grainIds to the parentIds
ebsd_merged('indexed').grainId = parentId(grains.id2ind(ebsd('indexed').grainId))
grains = calcGrains(ebsd_merged,'angle',10*degree)
plot(grains,grains.area)



But I got the same map.
![123](https://user-images.githubusercontent.com/106585808/171215231-cbeb0d00-2137-47fc-a4de-d3bcc57e88da.png)

What should I do to get the area map of the phase before twinning.
Thank you again for your help！！

**Reply by kilir** [reply]
What is your intention of the last invocation of `calcGrains()` after merging?

**Reply by gskgs** [reply]
I thought I could get the merge grains after  invoking  calcGrains().  I don't understand the function of this command correctly.

If  I use this code. 

%calculate the grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'threshold',10*degree);
%Copy the GB information to a new variable
gB = grains.boundary
%To not include the border of the EBSD map in calculations, create a new variable for only the mg-mg segments
gB_NiNi = gB('Ni-superalloy','Ni-superalloy')
%which can then be plotted, colorised by misorientation if you want
figure;plot(gB_NiNi,gB_NiNi.misorientation.angle./degree,'linewidth',2)
mtexColorbar

cs=ebsd ('Ni-superalloy').CS
%define an extension twin (60 degrees reorientation) by axis and rotation
%for magnesium extension twin
h= Miller(1,1,1, cs)
twinning = orientation('axis',h,'angle',60*degree,cs,cs)
%Next we create a logical of the boundary segments within 5 degrees of the twin relationship we just defined.
isTwinning = angle(gB_NiNi.misorientation,twinning) < 5*degree;
%locate the segments that meet this condition
twinBoundary = gB_NiNi(isTwinning)
% plot the grains, and overlay the twinning boundaries
figure;plot(grains,grains.meanOrientation)
hold on
plot(twinBoundary,'linecolor','blue','linewidth',2,'displayName','twin boundary')
hold off
%to calculate percentages of twin boundary out of all boundary segments
100*length(twinBoundary)/length(gB_NiNi)
[mergedGrains,parentId] = merge(grains,twinBoundary);
% plot the merged grains
hold on
plot(mergedGrains.boundary,'linecolor','k','linewidth',2.5,'linestyle','-',...
  'displayName','merged grains')
hold off

I can get this map.
![999](https://user-images.githubusercontent.com/106585808/171231437-ad1e28c1-48d2-4785-b802-29e86149ffd2.png)


I don't know what to do next to get the area map. If I write `plot(grains,grains.area)`,  I got the same map.
![123](https://user-images.githubusercontent.com/106585808/171230921-5bae5aee-1df4-4c9c-b76b-5caf906e5e2d.png)

I don't know how to get the area map of the grains before twinning.
Thank you very much for your reply

**Reply by kilir** [reply]
You should use the merged grains `mergedGrains` instead of `grains`.

Cheers,
Rüdiger

**Reply by gskgs** [reply]
> You should use the merged grains `mergedGrains` instead of `grains`.
> 
> Cheers, Rüdiger

I make it. Thanks for your help!!!
